### PR TITLE
Open edge from linux

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes # 
#3051 

**Changes proposed in this request**
Open microsoft-edge from linux

**Testing**
Tested manually from linux

**Performance impact**
None

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs [hunk 1]
```diff
@@ -172,25 +173,33 @@ public override Task StartDefaultOsBrowserAsync(string url)
...
-                                RedirectStandardError = true
-                            };
-
-                            Process.Start(psi);
-
-                            break;
-                        }
-                    }
 
+                    if (psi == null)
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:190]: isnt this always going to be null? because of line ?

**Review by trwalke — APPROVED** [review]
Looks good overall, just one question on the null check.

Seems like something that needs to be manually tested right? if so, lets add to the test sheet.

File: src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs [hunk 2]
```diff
@@ -90,7 +90,7 @@ internal interface IPlatformProxy
         /// <summary>
         /// Go to a URL using the OS default browser. 
         /// </summary>
-        Task StartDefaultOsBrowserAsync(string url);
+        Task StartDefaultOsBrowserAsync(string url, bool IsBrokerInstalled = false);
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs:93]:
```suggestion
        Task StartDefaultOsBrowserAsync(string url, bool isBrokerEnabled = false);
```

File: src/client/Microsoft.Identity.Client/Platforms/netcore/ApiConfig/SystemWebViewOptions.cs [hunk 3]
```diff
@@ -36,7 +36,7 @@ public static async Task OpenWithEdgeBrowserAsync(Uri uri)
             else
             {
                 var proxy = PlatformProxyFactory.CreatePlatformProxy(new NullLogger());
-                await proxy.StartDefaultOsBrowserAsync(url).ConfigureAwait(false);
+                await proxy.StartDefaultOsBrowserAsync(url, true).ConfigureAwait(false);
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/netcore/ApiConfig/SystemWebViewOptions.cs:39]: Is `true` also needs to be passed in on Line 65 below?

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs [hunk 4]
```diff
@@ -172,25 +173,33 @@ public override Task StartDefaultOsBrowserAsync(string url)
                     throw new MsalClientException(MsalError.LinuxXdgOpen, MsalErrorMessage.LinuxOpenAsSudoNotSupported);
                 }
 
+                List<string> openTools = new List<string>();
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:176]: Duplicate, unused, can be removed.

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs (hunk 1 above)
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:190]: This check is not needed, it's always true.

**Review by pmaytak — CHANGES_REQUESTED** [review]

File: src/client/Microsoft.Identity.Client/Platforms/Features/DefaultOSBrowser/DefaultOsBrowserWebUi.cs [hunk 5]
```diff
@@ -64,7 +64,8 @@ internal class DefaultOsBrowserWebUi : IWebUI
                 var authCodeUri = await InterceptAuthorizationUriAsync(
                     authorizationUri,
                     redirectUri,
-                    cancellationToken)
+                    cancellationToken,
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Features/DefaultOSBrowser/DefaultOsBrowserWebUi.cs:67]: by convention, the cancellation token is always the last param.

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs [hunk 6]
```diff
@@ -172,25 +173,33 @@ public override Task StartDefaultOsBrowserAsync(string url)
                     throw new MsalClientException(MsalError.LinuxXdgOpen, MsalErrorMessage.LinuxOpenAsSudoNotSupported);
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:173]: Can you check if you can start edge on Linux with `sudo` ? This exception prevents doing `sudo xdg-open` but maybe Edge is ok?

File: src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs:93]: or `isBrokerConfigured`

File: src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs [hunk 7]
```diff
@@ -90,7 +90,7 @@ internal interface IPlatformProxy
         /// <summary>
         /// Go to a URL using the OS default browser. 
         /// </summary>
-        Task StartDefaultOsBrowserAsync(string url);
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs:93]: Do we really need the param to have a default value? I find it easier to refactor code / reason about code when I'm asked to provide all params.

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
Small changes required. Main ask is to check if "sudo microsoft-edge url" works or not (it does not with xdg-open with Firefox for example. But maybe it works if invoking edge directly.

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs (hunk 6 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:173]: No edge does not open with sudo either

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs (hunk 4 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:176]: Missed to remove. Thanks for catching that

File: src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs (hunk 1 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs:190]: Missed to remove. Thanks

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 8]
```diff
@@ -145,7 +145,7 @@
 
   <ItemGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNetStandard)'">
     <Compile Include="$(PathToMsalSources)\Platforms\netstandard13\**\*.cs" LinkBase="Platforms\netstandard13" />
-    <Compile Include="$(PathToMsalSources)\Platforms\Features\DesktopOs\**\*.cs" />
+    <Compile Include="$(PathToMsalSources)\Platforms\Features\DesktopOS\**\*.cs" />
```
Comment (neha-bhargava) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:148]: To fix the build for linux as suggested by Bogdan.

File: src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs (hunk 7 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs:93]: Removed the default value

File: src/client/Microsoft.Identity.Client/Platforms/netstandard13/ApiConfig/SystemWebViewOptions.cs [hunk 9]
```diff
@@ -57,7 +57,7 @@ public static async Task OpenWithChromeEdgeBrowserAsync(Uri uri)
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
                 var proxy = PlatformProxyFactory.CreatePlatformProxy(new NullLogger());
-                await proxy.StartDefaultOsBrowserAsync(url).ConfigureAwait(false);
+                await proxy.StartDefaultOsBrowserAsync(url, false).ConfigureAwait(false);
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/netstandard13/ApiConfig/SystemWebViewOptions.cs:60]: Should be `true`?

**Review by pmaytak — APPROVED** [review]

**Review by bgavrilMS — APPROVED** [review]
LGTM, but please fix Peter's comments

**Pull request merged**